superset/assets/javascripts/modules/time.js

```diff
@@ -0,0 +1,54 @@
+const SECOND = 1000;  // milliseconds
+const MINUTE = 60 * SECOND;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
+const WEEK = 7 * DAY;
+const MONTH = 30 * DAY;
+const YEAR = 365 * DAY;
+const QUARTER = YEAR / 4;
+
+const unitsToMilliSeconds = {
+  second: SECOND,
+  minute: MINUTE,
+  hour: HOUR,
+  'half hour': HOUR / 2,  // mssql
+  day: DAY,
+  week: WEEK,
+  month: MONTH,
+  quarter: QUARTER,
+  year: YEAR,
+};
+
+const pattern = new RegExp('^(\\d+)?\\s*(.*)$');
+
+export const parseTimeGrain = function (timeGrain) {
```

Comment:
Wondering if we can have a single authority on time grain parsing between the frontend & backend.  Backend has `utils.parse_human_datetime` leveraging the `parsedatetime` lib. Just worried there's divergence into what the backend and frontend understand.

The backend could return its understanding of granularity to be available to the frontend in the payload. This may require a bit more thinking.

Reply:
Yeah, that would definitely be better. We can pass an [ISO 8601 duration](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) as a string. Let me take a look at it.